src/edu/csus/ecs/pc2/ui/ClarificationsTablePane.java

```diff
@@ -1114,7 +1130,24 @@ protected void requestSelectedClarification() {
         try {
             ElementId elementId = clarificationTable.getElementIdFromTableRow(selectedIndexes[0]);
             Clarification clarificationToAnswer = getContest().getClarification(elementId);
-
+            if (elementId.equals(lastAnswered)) {
+                // try to make it visible
+                answerClarificationFrame.setVisible(true);
+                return;
+            }
+            if (lastAnswered != null) {
+                int showConfirmDialog = JOptionPane.showConfirmDialog(getParentFrame(), "You are already answering a clar, do you want to return to that one (YES) or start answering the new one (NO)", "Already answering a Clarifition", JOptionPane.YES_NO_CANCEL_OPTION, JOptionPane.WARNING_MESSAGE, null);
```

Comment:
YES == Cancel, Maybe should only have Yes and Cancel; change Yes to switch to new clar Cancel keeps?

Reply:
I agree with JohnB; it works but I think the message to the user is slightly confusing.  For example, selecting "No" to cause it to move to the new clar is a bit unintuitive.

I would suggest what JohnB said:  change the dialog to have just two buttons, "Yes" and "Cancel", and change the dialog message to say:
```
You are asking to answer a new clarification, but you already have a clarification checked out for answering.  \n
If you want to give up the previous clarification and switch to answering the new one, click "Yes"; \n
otherwise, click "Cancel" and go back to answering the previous clarification.
```

Or an alternative would be to provide _three_ buttons:  `Switch to New Clar`, `Return to Previous Clar`, and `Cancel` (with appropriate adjustments to the suggested dialog message).

In either case I'd also like to see ToolTips added to the buttons explaining in more detail what they do.

Reply:
I'm not sure about the 3 button alternative.  What exactly would cancel do in that case?  Would it close the currently open clar and just return to the list of all clars, with none selected?  (That may surprise the user, since "cancel" is supposed to bring you back to where you were with things unchanged - in this case, it would CLOSE the already open answer clar dialog.  Not what you would expect.  Surprise!)   That's why the 2 button Yes/Cancel solution is probably the clearest.  IMHO.

Reply:
Perhaps you noticed I was using JOptionPane?  I can't control what the buttons say or their tooltips.
I can do a Yes/No solution.   With JohnC's message.